Code Review

segiddins opened the pull request:
This is a fake PR for the purposes of code review.

emkosz committed:
Source code for search moved from CocoaPods to this plugin. The plugin runs locally. 

Source code from Rainforest/CocoaPods/lib/cocoapods/command/search.rb to
cocoapods-search/lib/command/search.rb

Source code for specs also moved but require spec_helper in CocoaPods and can't run local atm.

emkosz committed:
Added specs and fixture files for specs

install the gem prettybacon
copied code from search_spec.rb in CocoaPods
added module Bacon in spec_helper.rb
require local files in spec_helper.rb
added description in gem spec

emkosz committed:
Most of the tests work now


Got rid of some debug prints in search.rb

emkosz committed:
All test works

Fixed indentation in search_spec.rb
Removed debug info in search.rb

emkosz committed:
Added travis.yml

kylef committed:
[README] Add description and usage information

kylef committed:
[README] Include a Travis CI badge

kylef committed:
[Web Search] Use SSL to connect to cocoapods.org

kylef committed:
Merge pull request #1 from CocoaPods/kylef/readme

Add description and usage information to the README

kylef committed:
Merge pull request #4 from CocoaPods/kylef/ssl

Use SSL to connect to cocoapods.org for web search

emkosz committed:
Added the original creators to the gem spec

emkosz committed:
Added the original creators to the gem spec

emkosz committed:
Added the creators to the license

emkosz committed:
Added a changelog

emkosz committed:
Correcting the changelog

emkosz committed:
Fixes to changelog

emkosz committed:
Correcting the changeling

emkosz committed:
Corrected the version in the CHANGELOG

segiddins commented on .travis.yml:
```diff
@@ -0,0 +1,9 @@
+branches:
+  only:
+    - master
+    - /.+-stable$/
```
newline

emkosz replied:
Where do I need, or need to get rid of a newline?

segiddins replied:
it's missing the trailing newline, which can lead to merge conflicts (and my text editor automatically dirtying the file)

segiddins commented on cocoapods-search.gemspec:
```diff
@@ -6,11 +6,13 @@ require 'cocoapods-search/gem_version.rb'
-  spec.description   = %q{A short description of cocoapods-search.}
-  spec.summary       = %q{A longer description of cocoapods-search.}
-  spec.homepage      = 'https://github.com/EXAMPLE/cocoapods-search'
+  spec.authors       = ['Eloy Durán','Fabio Pelosin','Emma Koszinowski']
```
space after `,`

emkosz replied:
this goes for the email addresses as well?

segiddins commented on lib/cocoapods-search/command/search.rb:
```diff
@@ -1,43 +1,97 @@
+        @full_text_search = argv.flag?('full')
+        @stats = argv.flag?('stats')
+        @supported_on_ios = argv.flag?('ios')
+        @supported_on_osx = argv.flag?('osx')
```
also need a `watchos` flag

kylef replied:
#6

segiddins commented on lib/cocoapods-search/command/search.rb:
```diff
@@ -1,43 +1,97 @@
+        @supported_on_ios = argv.flag?('ios')
+        @supported_on_osx = argv.flag?('osx')
+        @web = argv.flag?('web')
+        @query = argv.arguments! unless argv.arguments.empty?
```
the `unless` is unnecessary?

emkosz replied:
The test "complains for wrong parameters" fails without the unless. 
See terminal output:

Pod::Command::Search
  Search
    ✓ registers it self (11 ms)
    ✓ runs with correct parameters (21 ms)
    - complains for wrong parameters [FAILED]
    ✓ searches for a pod with name matching the given query ignoring case (3 ms)

segiddins replied:
Ah ok I see why now, we want the `query` to be nil. :+1:

segiddins commented on spec/spec_helper.rb:
```diff
@@ -5,41 +5,83 @@
+SpecHelper::Fixture.fixture('banana-lib') # ensure it exists
+SpecHelper::Fixture.fixture('orange-framework')
+
+
```
extraneous trailing newline

emkosz committed:
Fixes of some small issues

Added a trailing newline to travis.yml
Added spaces after commas in cocoapods-search.gemspec
Got rid of extraneous trailing newline in spec_helper.rb

emkosz committed:
Added a watchos flag

Tests are in the making.

emkosz committed:
Added test for watchos flag

Added a Realm.podspec file to run the watchos flag test

emkosz commented:
Fixes the issue https://github.com/CocoaPods/CocoaPods/issues/3397

emkosz committed:
Release 0.0.1

emkosz committed:
Release 0.1.0

emkosz committed:
Updated the information in README

emkosz committed:
Fixed the example outputs

emkosz committed:
Reedited the readme

orta committed:
Merge pull request #7 from CocoaPods/emkosz-cocoapods-search

cocoapods-search readme improvements.

manuyavuz committed:
Perform regexp escape on individual words before joining them

manuyavuz committed:
Command options description change.

manuyavuz committed:
Add specs and CHANGELOG

segiddins committed:
Merge pull request #10 from manuyavuz/imp/searchPerformance

Perform regexp escape on individual words before joining them

manuyavuz committed:
Add support for tvOS and any possible future platforms.

- Include specs for missing watchos cases, tvos, and any future platform.
- Update Gemfile.lock to cocoapods 0.39.0

manuyavuz committed:
Move all platforms logic to Core.

manuyavuz committed:
Point Gemfile to HEAD versions of cocoapods and core

segiddins committed:
Merge pull request #14 from manuyavuz/imp/supportAllFuturePlatforms

Add support for tvOS and any possible future platforms.

manuyavuz committed:
Perform full search as default, add `--simple` option to search only by name

segiddins committed:
Merge pull request #15 from manuyavuz/imp/fullAsDefault

Perform full text search as default.

manuyavuz committed:
Print output in reverse order

manuyavuz committed:
Add spec

segiddins committed:
Merge pull request #16 from manuyavuz/imp/switchOrdering

Print output in reverse order

segiddins committed:
Add .gitmodules
